go/tls/tls_test.go

```diff
@@ -0,0 +1,339 @@
+		{"also,bogus", nil, ""},
+		{"also,bogus:99", nil, ""},
+		{"maxval:1xyz", nil, ""},
+		{"maxval:1", &fieldInfo{count: 1, countSet: true}, ""},
+		{"maxval:255", &fieldInfo{count: 1, countSet: true}, ""},
+		{"maxval:256", &fieldInfo{count: 2, countSet: true}, ""},
+		{"maxval:65535", &fieldInfo{count: 2, countSet: true}, ""},
+		{"maxval:65536", &fieldInfo{count: 3, countSet: true}, ""},
+		{"maxval:16777215", &fieldInfo{count: 3, countSet: true}, ""},
+		{"maxval:16777216", &fieldInfo{count: 4, countSet: true}, ""},
+		{"maxval:16777216", &fieldInfo{count: 4, countSet: true}, ""},
+		{"maxval:4294967295", &fieldInfo{count: 4, countSet: true}, ""},
+		{"maxval:4294967296", &fieldInfo{count: 5, countSet: true}, ""},
+		{"maxval:1099511627775", &fieldInfo{count: 5, countSet: true}, ""},
+		{"maxval:1099511627776", &fieldInfo{count: 6, countSet: true}, ""},
+		{"maxval:281474976710655", &fieldInfo{count: 6, countSet: true}, ""},
+		{"maxval:281474976710656", &fieldInfo{count: 7, countSet: true}, ""},
+		{"maxval:72057594037927935", &fieldInfo{count: 7, countSet: true}, ""},
+		{"maxval:72057594037927936", &fieldInfo{count: 8, countSet: true}, ""},
+		{"minlen:1x", nil, ""},
+		{"maxlen:1x", nil, ""},
+		{"maxlen:1", &fieldInfo{count: 1, countSet: true, maxlen: 1}, ""},
+		{"maxlen:255", &fieldInfo{count: 1, countSet: true, maxlen: 255}, ""},
+		{"maxlen:65535", &fieldInfo{count: 2, countSet: true, maxlen: 65535}, ""},
+		{"minlen:65530,maxlen:65535", &fieldInfo{count: 2, countSet: true, minlen: 65530, maxlen: 65535}, ""},
+		{"maxlen:65535,minlen:65530", &fieldInfo{count: 2, countSet: true, minlen: 65530, maxlen: 65535}, ""},
+		{"minlen:65536,maxlen:65535", nil, "inverted"},
+		{"maxlen:16777215", &fieldInfo{count: 3, countSet: true, maxlen: 16777215}, ""},
+		{"maxlen:281474976710655", &fieldInfo{count: 6, countSet: true, maxlen: 281474976710655}, ""},
+		{"maxlen:72057594037927936", &fieldInfo{count: 8, countSet: true, maxlen: 72057594037927936}, ""},
+		{"size:0", nil, "unknown size"},
+		{"size:1", &fieldInfo{count: 1, countSet: true}, ""},
+		{"size:2", &fieldInfo{count: 2, countSet: true}, ""},
+		{"size:3", &fieldInfo{count: 3, countSet: true}, ""},
+		{"size:4", &fieldInfo{count: 4, countSet: true}, ""},
+		{"size:5", &fieldInfo{count: 5, countSet: true}, ""},
+		{"size:6", &fieldInfo{count: 6, countSet: true}, ""},
+		{"size:7", &fieldInfo{count: 7, countSet: true}, ""},
+		{"size:8", &fieldInfo{count: 8, countSet: true}, ""},
+		{"size:9", nil, "too large"},
+		{"size:1x", nil, ""},
+		{"size:1,val:9", nil, "selector value"},
+		{"selector:Bob,val:x9", &fieldInfo{selector: "Bob"}, ""},
+		{"selector:Fred,val:1", &fieldInfo{selector: "Fred", val: 1}, ""},
+		{"val:9,selector:Fred,val:1", &fieldInfo{selector: "Fred", val: 1}, ""},
+	}
+	for i, tc := range testData {
+		actual, err := fieldTagToFieldInfo(tc.tag, "")
+		if tc.errstr == "" {
+			if err != nil {
+				t.Errorf("#%d fieldTagToFieldInfo('%v') failed with %v", i, tc.tag, err)
```

Comment:
Done and done, passim.

https://golang.org/src/encoding/asn1/asn1_test.go#L68 (and many more)